pkgs/by-name/ya/yao-pkg/package.nix

```diff
@@ -0,0 +1,106 @@
+{
+  autoPatchelfHook,
+  fetchFromGitHub,
+  fetchYarnDeps,
+  fixup-yarn-lock,
+  lib,
+  makeWrapper,
+  nodejs,
+  stdenv,
+  yarn,
+  yarnBuildHook,
+  yarnConfigHook,
+  yarnInstallHook,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "pkg";
+  version = "6.1.1";
+
+  src = fetchFromGitHub {
+    owner = "yao-pkg";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-AQ+PVIWYgrflDsauxCfmvo40imlTAOW3vXhMtN+eKq4=";
+  };
+
+  offlineCache = fetchYarnDeps {
+    yarnLock = "${src}/yarn.lock";
+    hash = "sha256-7jIrhIuuHSkfQCnjifgYCEIHi169np78NLhuw6iZRyU=";
+  };
+
+  yarnBuildFlags = [
+    "--frozen-lockfile"
+    "--ignore-platform"
+    "--ignore-scripts"
+    "--no-progress"
+    "--offline"
+    "--prefer-offline"
+    "--production"
+    "--yarn-offline-mirror"
```

Comment:
Are *all* of these required? `--frozen-lockfile` and `--offline` at least should be set by default AFAIK

Reply:
I'm not sure tbh. I've looked at the docs and couldn't find much information on this and have looked at other submissions. I added these at the time just to be "extra safe" seeing this was my first nixpkg submission, although without them they also work? I've left them in but open to what you have to say about them if you think it needs to be removed

Reply:
You should try removing some of them and see if they still work :)